managers/asset-operations-manager.js

```diff
@@ -344,21 +344,46 @@ class AssetOperationsManager {
         );
 
         const { tokenId } = resolveUAL(UAL);
-        let hasPendingUpdate = false;
-        if (state === ASSET_STATES.LATEST) {
-            hasPendingUpdate = await this.blockchainService.hasPendingUpdate(tokenId, blockchain);
-        }
 
-        const publicAssertionId = hasPendingUpdate
-            ? await this.blockchainService.getUnfinalizedState(tokenId, blockchain)
-            : await this.blockchainService.getLatestAssertionId(tokenId, blockchain);
+        const unfinalizedState = await this.blockchainService.getUnfinalizedState(tokenId, blockchain);
+        const hasPendingUpdate = unfinalizedState !== ZeroHash;
+
+        let publicAssertionId;
+        let stateIsEnum = false;
+        if (hasPendingUpdate && ([ASSET_STATES.LATEST, unfinalizedState].includes(state))) {
+            publicAssertionId = unfinalizedState;
+            stateIsEnum = true;
+        } else {
+            const assertionIds = await this.blockchainService.getAssertionIds(tokenId, blockchain);
+
+            if (Object.values(ASSET_STATES).includes(state)) {
+                publicAssertionId = assertionIds[assertionIds.length - 1];
+                stateIsEnum = true;
+            } else if (typeof state === "number") {
+                if (state >= assertionIds.length) {
+                    return {
+                        errorType: 'DKG_CLIENT_ERROR',
+                        errorMessage: 'State index is out of range.',
+                    }
+                }
+
+                publicAssertionId = assertionIds[state];
+            } else if (assertionIds.includes(state)) {
+                publicAssertionId = state;
+            } else {
+                return {
+                    errorType: 'DKG_CLIENT_ERROR',
+                    errorMessage: 'Incorrect state option.',
+                }
+            }
+        }
 
         const getPublicOperationId = await this.nodeApiService.get(
             endpoint,
             port,
             authToken,
             UAL,
-            state,
+            stateIsEnum ? state: publicAssertionId,
```

Comment:
can't this just be Object.values(ASSET_STATES).includes(state) ? state : publicAssertionId